scripts/o3de/tests/test_compatibility.py

```diff
@@ -181,3 +181,47 @@ def test_resolve_gem_dependencies(gem_names, all_gem_json_data, expected_result)
     else:
         assert not expected_result
         assert errors
+
+@pytest.mark.parametrize(
+    "num_root_gems, num_dependencies_per_gem, num_optional_gems, tree_depth", [
+        # Given:
+        # R num_root_gems
+        # N num_dependencies_per_gem
+        # D tree_depth - 1
+        # Total gems = R * (pow(N, D + 1) - 1) / (N -1)
```

Comment:
If the root of the tree is defined as level 0, then the `D + 1` part of the algorithm becomes just `D`

Reply:
oh yeah, good point, will update